specs/lib/gltfPipelineSpec.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+var clone = require('clone');
+var gltfPipeline = require('../../lib/gltfPipeline');
+var processJSON = gltfPipeline.processJSON;
+var processFile = gltfPipeline.processFile;
+var readGltf = require('../../lib/readGltf');
+var writeBinaryGltf = require('../../lib/writeBinaryGltf');
+
+var gltfPath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest.gltf';
+var glbPath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest.glb';
+
+describe('gltfPipeline', function() {
+    it('optimizes a gltf file', function(done) {
+        var gltfCopy;
+        var options = {};
+        readGltf(gltfPath, function(gltf) {
+            gltfCopy = clone(gltf);
+            processJSON(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+                done();
+            });
+        });
+        processFile(gltfPath, options, function(gltf) {
+            expect(gltf).toBeDefined();
+            expect(gltf).not.toBe(gltfCopy);
+            done();
+        })
+    });
+    
+    it('optimizes a binary glTF file', function(done) {
+        var gltfCopy;
+        var options = {};
+        readGltf(glbPath, function(gltf) {
+            gltfCopy = clone(gltf);
+            processJSON(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+                done();
+            });
+        });
+        processFile(glbPath, options, function(gltf) {
+            expect(gltf).toBeDefined();
+            expect(gltf).not.toBe(gltfCopy);
+            done();
+        });
+    });
+});
```

Comment:
This file tests `processFile`, `processJson`, but we should also test all the other functions in `gltfPipeline`. Also in order to test the write function (json and binary), we can spy on `fs.writeFile` and check that the outputPath argument is correct. (Check out this for an example: https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Specs/Scene/ShadowMapSpec.js#L948-L967, but don't use `callThrough` because we don't want to save the file).